File: extract_ro1.js
const fs = require('fs');
const ts = require('typescript');

const code = fs.readFileSync('lib/visualLab/data/poiExtraRo1.ts', 'utf8');
const result = ts.transpileModule(code, { compilerOptions: { module: ts.ModuleKind.CommonJS } });
fs.writeFileSync('.tmp_ro1.js', result.outputText);

const pois = require('./.tmp_ro1.js').poiExtraRo1;
fs.writeFileSync('ro1_pois.json', JSON.stringify(pois, null, 2));

File: extract_ro1.py
import json
import re

with open('/mnt/c/Users/User/plizio-repo/lib/visualLab/data/poiExtraRo1.ts') as f:
    text = f.read()

pois = []
matches = re.finditer(r'\{[^\}]*?\"id\":\s*\"([^\"]+)\".*?\"name\":\s*\{[^\}]*?\"ro\":\s*\"([^\"]*)\"[^\}]*?\}.*?\"description\":\s*\{[^\}]*?\"ro\":\s*\"([^\"]*)\"', text, re.DOTALL)
for m in matches:
    pois.append({
        "id": m.group(1),
        "name_ro": m.group(2),
        "desc_ro": m.group(3)
    })

with open('/mnt/c/Users/User/plizio-repo/ro1_pois.json', 'w') as f:
    json.dump(pois, f, indent=2)

print(f"Extracted {len(pois)} POIs")
